Add JSON serialization and deserialization to Position and RoutePoint classes

quinton-hoole opened the pull request:
This is in preparation for further testing.  It is necessary to be able to capture object state and sensible inputs to routing algorithm in order to build re-usable test data. I used JSON just because we already use the libraries, it's convenient, and easy to hand-edit should it be needed. I also wrote some more unit tests to test the new serde methods. 

All commits before "[Add json serialization to Position and RoutePoint classes](https://github.com/rgleason/weather_routing_pi/commit/2b280737552d358e18661a8d3fa0424645250a96) can be ignored, as they are part of separate PR's that will be reviewed and merged indepently, and then disappear from this PR after rebasing.

```
make test
...
      Start 62: PositionTest.ConstructorBasic
62/83 Test #62: PositionTest.ConstructorBasic .................   Passed    0.02 sec
      Start 63: PositionTest.toJsonBasic
63/83 Test #63: PositionTest.toJsonBasic ......................   Passed    0.02 sec
      Start 64: PositionTest.fromJsonBasic
64/83 Test #64: PositionTest.fromJsonBasic ....................   Passed    0.02 sec
      Start 65: RoutePointTest.ConstructorBasic
65/83 Test #65: RoutePointTest.ConstructorBasic ...............   Passed    0.02 sec
      Start 66: RoutePointTest.toJsonBasic
66/83 Test #66: RoutePointTest.toJsonBasic ....................   Passed    0.02 sec
      Start 67: RoutePointTest.fromJsonBasic
67/83 Test #67: RoutePointTest.fromJsonBasic ..................   Passed    0.03 sec
...
100% tests passed, 0 tests failed out of 83
```

```
make coverage
...
File 'Position.cpp'
Lines executed:8.52% of 317
...
File 'RoutePoint.cpp'
Lines executed:10.09% of 337
...
```


cc @sebastien-rosset @rgleason

sebastien-rosset commented on test/RoutePoint_tests.cpp:
```diff
@@ -0,0 +1,44 @@
+    EXPECT_EQ(m_routePoint.grib_is_data_deficient, m_data_deficient);
+}
+
+TEST_F(RoutePointTest, toJsonBasic) {
```
For test naming conventions, see:
 
- https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#follow-test-naming-standards
- https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea
- https://www.reddit.com/r/cpp/comments/1046xnv/how_do_you_name_your_unit_tests/
- https://dzone.com/articles/7-popular-unit-test-naming
- https://khorikov.org/posts/2021-02-25-test-naming-conventions/
- https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html

Anti-patterns:
- https://stackoverflow.com/questions/155436/unit-test-naming-best-practices
- https://medium.com/@rostyslav.ivankiv/dont-make-these-java-unit-testing-mistakes-and-avoid-anti-patterns-3987a1b81a5c
- https://stackoverflow.com/questions/333682/unit-testing-anti-patterns-catalogue

sebastien-rosset reviewed (changes requested):
Same comments as #448, with regards to unit test naming conventions and chasing high LoC metrics versus meaningful algorithmic coverage.

rgleason commented:
Quinton, also we are trying to work out of "master" now in accord with Sebastien's request.  I can't seem to change where the PR is pointed in the online interface.

quinton-hoole commented:
I can rebase and squash this PR to a single commit as soon as #449 has been merged (and that one is ready to merge, but there seems to be some problem related to signed commits being required).

quinton-hoole commented:
Yes, indeed. That is what I offered to do. I will do so later today, after sailing to our new motu :-)

quinton-hoole commented:
@rgleason I have rebased, squashed to a single commit, and retested.  This is ready to merge now.

```
 % make test
Running tests...
100% tests passed, 0 tests failed out of 78

Total Test time (real) =   1.83 sec
```
